erpnext/accounts/doctype/payment_entry/payment_entry.js

```diff
@@ -140,6 +140,11 @@ frappe.ui.form.on('Payment Entry', {
 		});
 	},
 
+	before_submit: (frm) => {
+		if (!frm.doc.reference_no || !frm.doc.reference_date)
+			frm.toggle_reqd(['reference_no', 'reference_date'], 1);
+	},
+
```

Comment:
@bhattdevarsh this is not required. as of now you are handling it to server side.
```suggestion
	
```